crates/uv-distribution-types/src/status_code_strategy.rs

```diff
@@ -0,0 +1,205 @@
+use http::StatusCode;
+use rustc_hash::FxHashSet;
+use url::Url;
+
+use crate::{IndexCapabilities, IndexUrl};
+
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub enum IndexStatusCodeStrategy {
+    #[default]
+    Default,
+    IgnoreErrorCodes {
+        status_codes: FxHashSet<StatusCode>,
+    },
+}
+
+impl IndexStatusCodeStrategy {
+    /// Derive a strategy from an index URL. We special-case PyTorch. Otherwise,
+    /// we follow the default strategy.
+    pub fn from_index_url(url: &Url) -> Self {
+        if url
+            .host_str()
+            .is_some_and(|host| host.contains("pytorch.org"))
+        {
+            // The PyTorch registry returns a 403 when a package is not found, so
+            // we ignore them when deciding whether to search other indexes.
+            Self::IgnoreErrorCodes {
+                status_codes: FxHashSet::from_iter(vec![StatusCode::FORBIDDEN]),
+            }
+        } else {
+            Self::Default
+        }
+    }
+
+    /// Derive a strategy from a list of status codes to ignore.
+    pub fn from_ignored_error_codes(status_codes: &[u16]) -> Self {
+        let status_codes = status_codes
+            .iter()
+            .map(|code| StatusCode::from_u16(*code))
+            .collect::<Result<FxHashSet<_>, _>>()
+            .expect("Status codes should be valid.");
```

Comment:
Why `expect` here? How do we know they're valid? Should this just return an `Err`?